scheduler/src/workflow-execution.ts

```diff
@@ -0,0 +1,114 @@
+import * as AdapterClient from './clients/adapter-client'
+import * as CoreClient from './clients/core-client'
+import * as TransformationClient from './clients/transformation-client'
+import * as StorageClient from './clients/storage-client'
+
+import DatasourceConfig from './interfaces/datasource-config'
+import PipelineConfig from './interfaces/pipeline-config'
+import NotificationConfig from './interfaces/notification-config'
+import { AxiosError } from 'axios'
+import AdapterResponse from '@/interfaces/adapter-response'
+
+export async function execute (datasourceConfig: DatasourceConfig, maxRetries = 3): Promise<void> {
```

Comment:
Added a function `retryableExecution` that retries each step instead of whole workflow. Was influencing system tests too heavily ....

Reply:
Very nice!